pkg/tfbridge/validate_input_types.go

```diff
@@ -35,215 +35,249 @@ func NewInputValidator(urn resource.URN, schema pschema.PackageSpec) *PulumiInpu
 	}
 }
 
-// getObjectTypeSpec gets a type spec for an object type. Objects can have their
-// type defined in a couple different ways. This will check the different ways
-// and return the correct type spec.
-func (v *PulumiInputValidator) getObjectTypeSpec(
-	spec *pschema.TypeSpec,
-	propertyValue resource.PropertyValue,
-) (
-	complexSpec *pschema.ComplexTypeSpec,
-	typeSpec *pschema.TypeSpec,
-	requiredProps []string,
-) {
-
-	if spec.AdditionalProperties != nil {
-		if spec.AdditionalProperties.Ref != "" {
-			complexSpec = v.getType(spec.AdditionalProperties.Ref)
-			// if we don't find the reference then just continue instead of failing
-			if complexSpec != nil {
-				requiredProps = complexSpec.Required
+// validatePropertyValue is the main function for validating a PropertyMap against the Pulumi Schema. It is a
+// recursive function that will validate nested types and arrays. Returns a list of type failures if any are found.
+func (v *PulumiInputValidator) validatePropertyMap(
+	propertyMap resource.PropertyMap,
+	required []string,
+	propertyTypes map[string]pschema.PropertySpec,
+	propertyPath resource.PropertyPath,
+) []TypeFailure {
+	stableKeys := propertyMap.StableKeys()
+	failures := []TypeFailure{}
+
+	// handle required properties
+	//
+	// TODO: what if properties have defaults specified, should this still be an error?
```

Comment:
Actually important to dig into this TODO, I think we might be throwing extraneous warnings otherwise, how things stand, where bridge would actually auto-fill the default post-Check. Pretty sure that's the case.. so perhaps just not validate these.